src/reader_util.cpp

```diff
@@ -374,3 +376,19 @@ std::string ReaderUtil::Recode(const std::string& str_to_encode,
 	return result;
 #endif
 }
+
+std::string ReaderUtil::Normalize(const std::string &str) {
+#ifdef LCF_SUPPORT_ICU
+	icu::UnicodeString& uni = icu::UnicodeString(str.c_str()).toLower();
+	UErrorCode err = U_ZERO_ERROR;
+	const icu::Normalizer2* norm = icu::Normalizer2::getNFKCInstance(err);
+	icu::UnicodeString f = norm->normalize(uni, err);
```

Comment:
Since ICU will not touch the destination string in error case, maybe we should return the former string?
```c++
if (U_FAILURE(err))
```

Reply:
added return of lowercase string in error case and removed the &-ref, that looked like a bug.